api/handlers/metadata.go

```diff
@@ -23,46 +23,67 @@ import (
 // GetMetadataHandlerFunc returns metadata of the keptn installation
 func GetMetadataHandlerFunc(params metadata.MetadataParams, principal *models.Principal) middleware.Responder {
 
+	handler := newMetadataHandler()
+
+	return handler.getMetadata()
+}
+
+func newMetadataHandler() metadataHandler {
+	var clientSet kubernetes.Interface
 	logger := keptnutils.NewLogger("", "", "api")
-	logger.Info("API received a GET metadata event")
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logger.Debug("Could not get InClusterConfig, will skip k8s-deployments.")
+	} else {
+		// creates the clientset
+		clientSet, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			logger.Debug("Could not create kubernetes client, will skip k8s-deployments.")
+		}
+	}
+	return metadataHandler{
+		k8sClient:       clientSet,
+		logger:          logger,
+		swaggerFilePath: "/swagger-ui/swagger.yaml",
+	}
```

Comment:
What is returned here if `clientSet` is not set because of an error when retrieving the inClusterConfig?